src/julia_gc.c

```diff
@@ -801,9 +799,6 @@ void GAP_InitJuliaMemoryInterface(jl_module_t *   module,
         parent = jl_any_type;
     }
 
-// Julia defines HAVE_JL_REINIT_FOREIGN_TYPE if `jl_reinit_foreign_type`
-// is available.
-#ifdef HAVE_JL_REINIT_FOREIGN_TYPE
     if (gap_jl_boundp(module, jl_symbol("GapObj"))) {
```

Comment:
are there use-cases where this evaluates to false or is this another artifact from the time where both GAP and GAP.jl adapted to the reinit_foreign_type things, but wanted to stay compatible with older versions of each other? If the former, we could remove the check, inline the block content, drop everything below, and also drop some helper functions just used in the else case.

I found one place in yggdrasil, but we can get rid of it with https://github.com/JuliaPackaging/Yggdrasil/pull/11697

Reply:
Yes it is a left-over from the times before reinit_foreign_type etc., so it could be simplified now.

Reply:
I further simplified this in [`c6ad70a` (#6043)](https://github.com/gap-system/gap/pull/6043/commits/c6ad70af0d09fec20bbb5e220738751657a7bf9d). According to CI, GAP.jl is happy with this change, but just using the julia gc (i.e. `--with-gc=julia --with-julia=/path/to/julia`) segfaults...
My best guess is that in that case we call https://github.com/gap-system/gap/blob/c6ad70af0d09fec20bbb5e220738751657a7bf9d/src/julia_gc.c#L813 which due to the first arg being 0 defaults to using the julia `Main` module, but there the expected type declarations are missing.
Until now, I didn't even know that there was a way to run gap-with-julia but without GAP.jl, so I have no idea how to proceed with this.

Reply:
Ok. So to make progress on this PR here, I will just revert the second commit. Getting rid of this configure option and corresponding CI test should IMO be its separate PR.